ocaml/xapi/vm_evacuation.ml

```diff
@@ -34,6 +34,10 @@ let ensure_no_vms ~__context ~rpc ~session_id ~evacuate_timeout =
     get_resident_vms ~__context ~self:host |> snd
     |> List.filter (fun vm -> is_running vm && not (self_managed_poweroff vm))
   in
+  let get_driver_domains () =
+    get_resident_vms ~__context ~self:host |> fst
+    |> List.filter (fun vm -> is_running vm && not (Db.VM.get_is_control_domain ~__context ~self:vm))
```

Comment:
to me this filter reads as 'all running domUs', but i imagined your driver domains would be a strict subset of your domUs?

Reply:
ah i understand from `get_resident_vms` why this works... that name is not the best

Reply:
Yes, the use of `fst` and `snd` here is confusing, I'll use a let binding instead to make it more explicit what the 2 sides mean.